src/VS.Web.CG.Mvc/RazorPage/RazorPageScaffolderBase.cs

```diff
@@ -284,7 +284,7 @@ protected RazorPageWithContextTemplateModel GetRazorPageWithContextTemplateModel
 
         private string GetDefaultPageModelNamespaceName(string relativeFolderPath)
         {
-            return NameSpaceUtilities.GetSafeNameSpaceFromPath(relativeFolderPath, ApplicationInfo.ApplicationName);
+            return NameSpaceUtilities.GetSafeNameSpaceFromPath(relativeFolderPath, _projectContext.RootNamespace);
```

Comment:
Is this used in commandline cases & when VS does not pass the namespace?

Reply:
VS should always pass a namespace but yes in cmd line cases.